File: set1/4/main.py
import sys

def input():
    return sys.stdin.readline().rstrip()

N, X = map(int, input().split())
arr = list(map(int, input().split()))

s = sum(arr[:X])
mx, cnt = s, 1

for i in range(X, N):
    s -= arr[i - X]
    s += arr[i]
    if mx < s:
        mx = s
        cnt = 1
    elif mx == s:
        cnt += 1

print("SAD" if mx == 0 else f"{mx} {cnt}")

File: set1/4/main.cpp
#include<iostream>
#include<vector>

using namespace std;

int main(){
    ios::sync_with_stdio(false);
    cin.tie(0);

    int N, X; cin >> N >> X;
    vector<int> arr(N);
    for(int i = 0; i < N; ++i) cin >> arr[i];

    int sum = 0;
    for(int i = 0; i < X; ++i) sum += arr[i];
    int mx = sum, cnt = 1;

    for(int i = X; i < N; ++i) {
        sum -= arr[i - X];
        sum += arr[i];
        if(mx < sum) {
            mx = sum;
            cnt = 1;
        }
        else if(mx == sum) ++cnt;
    }
    if(mx == 0) cout << "SAD";
    else cout << mx << '\n' << cnt;

    return 0;
}